Calendly: Removed look behinds in regexes

pablinos opened the pull request:
#### Changes proposed in this Pull Request:
Look behinds aren't supported by [Firefox, IE and some other browsers](https://caniuse.com/#feat=js-regexp-lookbehind), which prevented the Calendly block from registering correctly. 

This change replaces those regular expressions with ones that use groups to grab what is needed from the string.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
This fixes a bug with the soon to be launched Calendly block

#### Testing instructions:
With a Jetpack site, activate the beta blocks and test that the different types of embed code can still be used with the Calendly block.

Optionally test it in another browser that doesn't support look behinds e.g. Firefox.

#### Proposed changelog entry for your changes:
* N/A

pablinos committed:
Removed look behinds in regexes

This isn't supported by Firefox, IE and possibly other browsers

pablinos commented:
Fixes #14376

jeherve reviewed (changes requested):
Tests appear to fail now:
https://travis-ci.org/Automattic/jetpack/jobs/638156137?utm_medium=notification&utm_source=github_status
https://github.com/Automattic/jetpack/blob/2c3dbd93a0cc735df5593a147532cf5de02627ba/extensions/blocks/calendly/test/utils.js#L44

pablinos committed:
Fixed the regex for the failing bare URLs test

The new regex wasn't taking into account the case where the embed code
could be a bare URL starting with `calendly.com`

pablinos commented:
Whoops! Thanks @jeherve. That should be fixed now.

jeherve reviewed (approved):
Let's give this a try. It seems to work on my end!
